analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -416,35 +384,23 @@ def behavior(self):
 def get_query(source: ObjectStream) -> ObjectStream:
     """Query for event / column selection: >=4j >=1b, ==1 lep with pT>25 GeV, return relevant columns
     """
-    return source.Where(lambda e:
-        # == 1 lep
-        e.electron_pt.Where(lambda pT: pT > 25).Count() + e.muon_pt.Where(lambda pT: pT > 25).Count()== 1
-        )\
-        .Where(lambda e:\
-            # >= 4 jets
-            e.jet_pt.Where(lambda pT: pT > 25).Count() >= 4
-        )\
-        .Where(lambda e:\
-            # >= 1 jet with pT > 25 GeV and b-tag >= 0.5
-            {"pT": e.jet_pt, "btag": e.jet_btag}.Zip().Where(lambda jet: jet.btag >= 0.5 and jet.pT > 25).Count() >= 1
-        )\
-        .Select(lambda e:\
-            # return columns
-            {
-                "electron_e": e.electron_e,
-                "electron_pt": e.electron_pt,
-                "muon_e": e.muon_e,
-                "muon_pt": e.muon_pt,
-                "jet_e": e.jet_e,
-                "jet_pt": e.jet_pt,
-                "jet_eta": e.jet_eta,
-                "jet_phi": e.jet_phi,
-                "jet_btag": e.jet_btag,
-                "numbermuon": e.numbermuon,
-                "numberelectron": e.numberelectron,
-                "numberjet": e.numberjet,
-            }
-        )
+    return source.Where(lambda e: e.Electron_pt.Where(lambda pt: pt > 25).Count() 
+                        + e.Muon_pt.Where(lambda pt: pt > 25).Count() == 1)\
+                 .Where(lambda e: e.Jet_pt.Where(lambda pt: pt > 25).Count() >= 4)\
+                 .Where(lambda g: {"pt": g.Jet_pt, 
+                                   "btagCSVV2": g.Jet_btagCSVV2}.Zip().Where(lambda jet: 
+                                                                             jet.btagCSVV2 >= 0.5 
+                                                                             and jet.pt > 25).Count() >= 1)\
+                 .Select(lambda f: {"Electron_pt": f.Electron_pt,
+                                    "Electron_mass": f.Electron_mass,
+                                    "Muon_pt": f.Muon_pt,
+                                    "Muon_mass": f.Muon_mass,
+                                    "Jet_mass": f.Jet_mass,
```

Comment:
Do we need the masses here specifically or does the energy also work?

Reply:
Energy did not work. Maybe because the energy is not a branch in the nanoAOD (It is calculated when running the NanoAODSchema)